src/main/java/com/internet/shop/dao/jdbc/impl/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,173 @@
+package com.internet.shop.dao.jdbc.impl;
+
+import com.internet.shop.dao.ShoppingCartDao;
+import com.internet.shop.exceptions.DataProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Product;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ShoppingCartDaoJdbcImpl implements ShoppingCartDao {
+    @Override
+    public Optional<ShoppingCart> getByUserId(Long userId) {
+        String query = "SELECT * FROM shopping_carts WHERE user_id = ? AND isDeleted = FALSE";
+        Optional<ShoppingCart> neededCart = Optional.empty();
```

Comment:
I think it will be better to first initialise `ShoppingCart neededCart = null;`
then set products `if (neededCart != null)`
and return Optional.ofNullable(neededCart)